### PR TITLE
entgql: preload edge with alias

### PR DESCRIPTION
**Pull request by giautm**
Issue: https://github.com/ent/ent/issues/2755

### CODE REVIEW

File: entgql/extension.go [hunk 1]
```diff
@@ -208,14 +208,16 @@ func (e *Extension) Hooks() []gen.Hook {
 func (e *Extension) genSchemaHook() gen.Hook {
 	return func(next gen.Generator) gen.Generator {
 		return gen.GenerateFunc(func(g *gen.Graph) (err error) {
+			hasNamedEdges, _ := g.FeatureEnabled(gen.FeatureNamedEdges.Name)
+			if !hasNamedEdges {
+				return fmt.Errorf("entgql: the entgql extension requires the `namedges` feature")
+			}
```
Comment (a8m) [inline entgql/extension.go:214]: I'd say, let's enable it for the user in case it's not enabled.

File: entgql/extension.go [hunk 2]
```diff
@@ -208,14 +208,16 @@ func (e *Extension) Hooks() []gen.Hook {
...
 			if err = next.Generate(g); err != nil {
 				return err
 			}
-
```
Comment (a8m) [inline entgql/extension.go:214]: 😍

File: entgql/extension.go [hunk 3]
```diff
@@ -208,14 +208,16 @@ func (e *Extension) Hooks() []gen.Hook {
...
 			if e.path == "" || !(e.genSchema || e.genWhereInput || e.genMutations) {
 				return nil
 			}
-
```
Comment (a8m) [inline entgql/extension.go:218]: 😍

File: entgql/template/edge.tmpl [hunk 4]
```diff
@@ -59,27 +59,25 @@ import "context"
 		{{- end }}
 		}
 		{{- /* May be nil if the totalCount was not loaded. */}}
-		totalCount := {{ $r }}.Edges.totalCount[{{ $i }}]
+		alias := graphql.GetFieldContext(ctx).Field.Alias
```
Comment (a8m) [inline entgql/template/edge.tmpl:62]: 💯

File: entgql/template/collection.tmpl [hunk 5]
```diff
@@ -109,7 +113,15 @@ func ({{ $receiver }} *{{ $query }}) collectField(ctx context.Context, op *graph
...
-						{{ $receiver }}.{{ $e.EagerLoadField }} = query
+						{{- if $e.Unique }}
+						{{ $receiver }}.With{{ $e.StructField }}(func (wq *{{ $e.Type.QueryName }}) {
+							*wq = *query
+						})
+						{{- else }}
+						{{ $receiver }}.WithNamed{{ $e.StructField }}(alias, func (wq *{{ $e.Type.QueryName }}) {
+							*wq = *query
+						})
+						{{- end }}
```
Comment (a8m) [inline entgql/template/collection.tmpl:124]: Why not?
```suggestion
						{{- if $e.Unique }}
							{{ $receiver }}.{{ $e.EagerLoadField }} = query
						{{- else }}
							{{ $receiver }}.WithNamed{{ $e.StructField }}(alias, func (wq *{{ $e.Type.QueryName }}) {
								*wq = *query
							})
						{{- end }}
```

**Review by a8m — APPROVED** [review]
🚀

**Review by a8m — APPROVED** [review]

**Comment by a8m** [comment]
Excellent @giautm ❤️

**Comment by giautm** [comment]
> Excellent @giautm ❤️

Thank you for your work on https://github.com/ent/ent/pull/2792, <3

**Pull request merged**